### DISCUSSION TITLE
starcoin 发布智能合约。

### DISCUSSION THREAD

**Discussion by v1xingyue** [NonceGeek/Move-Camp#186] [Materials]
大家开发的时候，基本上都是在本地的dev 环境完成部署。

但是，当开发完成需要发布的时候，需要怎么做呢？

目前有两个选择: 

1. 通过 [movetools](https://movetool.app/)  上传编译好的文件，完成部署。 
2. 通过console 连接到 远程节点，使用dev 部署。

连接的地址: ws://main.seed.starcoin.org:9870 
注意: 连接远程的rpc 节点需要初始化一个本地目录。
这个目录是用来放置你本地的账号信息的。部署合约也是通过这个账号。 同时会从这个账号扣除相应的gas 费。
如果你想部署到你自己原来的账号里，需要通过 starmask 导出对应的账号的私钥。 
然后通过 account import 导入 即可。 

简单流程如下: 

1. starcoin --local-account-dir ./wallet --connect ws://main.seed.starcoin.org:9870 console  
    starcoin --local-account-dir ./wallet --connect ws://barnard.seed.starcoin.org:9870 console 
3. account import  -i key  -p  pass 
4. dev deploy 

完事儿，你的合约就跑在真实的线上环境了。

**Comment by leeduckgo** [comment]
Great.
